File: leetcode/1/1.js
var twoSum = function(nums, target) {
    var table = {};
    for( i in nums ){
      var tmp = table[ target - nums[i] ]
      if( tmp < nums.length && tmp >= 0)
        return [parseInt(tmp), parseInt(i)]           
      else
        table[nums[i]] = i
      //console.log(i)
      //console.log(target-nums[i])
      //console.log(tmp)
    }
    return "no answer";
};

File: leetcode/1/1.cpp
#include <iostream>
#include <unordered_map>
#include <vector>
#include <string>

using namespace std;

class Solution {
  public:
    vector<int> twoSum(vector<int>& nums, int target) {
      std::unordered_map<int, int> table;
      for(int i = 0; i < nums.size(); i++){
        if(table.find(target - nums[i]) != table.end())
          return vector<int>{table[target - nums[i]], i};
        else
          table.insert({nums[i], i});
        //std::cout << i << std::endl;
        //std::cout << target - nums[i] << std::endl;
        //std::cout << ( (table.find(target - nums[i]) != table.end()) == true ? "true" : "false" )<< std::endl;
      }
      return {};
    }
};

int main(){
  Solution a;
  vector<int> b = {3,2,4};
  a.twoSum (b, 6);
}

